commodore/dependency_mgmt.py

```diff
@@ -8,6 +9,9 @@
 from .helpers import yaml_load
 
 
+Component = collections.namedtuple('Component', 'name repository_url target_directory')
```

Comment:
Consider reusing/extending `config.Component`, which is already used to track component Git URLs and desired versions.